File: add_n_num.py
def getdata(num):
    n=int(input("Enter size of num:"))
    print("Enter the num:")
    for i in range(n):
        x=int((input()))
        num.append(x)
        i+=1
    return num
    
def compute(num):
        mysum=0
        n=len(num)
        mysum=sum(num)
        #print("num")
        return mysum
    
def output(mysum):
    print("The sum of n number is :",mysum)
     
def main():

    arr=[]
    arr=getdata(arr)
    result=compute(arr)
    output(result)

main()

File: add_n_num.c
#include<stdio.h>
void getdata(int *a,int *n)
{       int i;
        printf("Enter the size of num:");
        scanf("%d",n);
        printf("Enter the numbers:");
        for(i=0;i<*n;i++)
        {
                scanf("%d",a);
                a++;
        }

}
void add(int *a,int n,int *sum)
{       int i;

        for(i=0;i<n;i++)
        {
                *sum+=*a;
                a++;
        }
}

void putdata(int sum)
{
                printf("The sum of n numbers is %d ",sum);
}
int main()
{

        int a[10],n,*p,sum=0;
        p=a;
        getdata(p,&n);
        add(p,n,&sum);
        putdata(sum);


}

~
~
~
~
